src/OSLikeStuff/task_scheduler.cpp

```diff
@@ -152,7 +174,7 @@ void TaskManager::runTask(TaskID id) {
 	}
 }
 void TaskManager::clockRolledOver() {
-	for (int i = 0; i < index; i++) {
+	for (int i = 0; i < numActiveTasks; i++) {
```

Comment:
shouldn't this also loop over all tasks, since if I understand correctly `list` can have holes?

Reply:
Yeah it should